lib/collection/src/operations/conversions.rs

```diff
@@ -132,6 +137,143 @@ impl From<Record> for api::grpc::qdrant::RetrievedPoint {
+                            ))
+                        }
+                        Some(distance) => distance,
+                    },
+                    shard_number: NonZeroU32::new(params.shard_number).unwrap(),
+                },
+            },
+            hnsw_config: match config.hnsw_config {
+                None => return Err(Status::invalid_argument("Malformed HnswConfig type")),
+                Some(hnsw_config) => hnsw_config.into(),
+            },
+            optimizer_config: match config.optimizer_config {
+                None => return Err(Status::invalid_argument("Malformed OptimizerConfig type")),
+                Some(optimizer_config) => optimizer_config.into(),
+            },
+            wal_config: match config.wal_config {
+                None => return Err(Status::invalid_argument("Malformed WalConfig type")),
+                Some(wal_config) => wal_config.into(),
+            },
+        })
+    }
+}
+
+impl TryFrom<api::grpc::qdrant::GetCollectionInfoResponse> for CollectionInfo {
+    type Error = Status;
+
+    fn try_from(
+        collection_info_response: api::grpc::qdrant::GetCollectionInfoResponse,
+    ) -> Result<Self, Self::Error> {
+        match collection_info_response.result {
+            None => Err(Status::invalid_argument("Malformed CollectionInfo type")),
+            Some(collection_info_response) => {
+                Ok(Self {
+                    status: collection_info_response.status.try_into()?,
+                    optimizer_status: match collection_info_response.optimizer_status {
+                        None => {
+                            return Err(Status::invalid_argument("Malformed OptimizerStatus type"))
+                        }
+                        Some(api::grpc::qdrant::OptimizerStatus { ok, error }) => {
+                            if ok {
+                                OptimizersStatus::Ok
+                            } else {
+                                OptimizersStatus::Error(error)
+                            }
+                        }
+                    },
+                    vectors_count: collection_info_response.vectors_count as usize,
+                    segments_count: collection_info_response.segments_count as usize,
+                    disk_data_size: collection_info_response.disk_data_size as usize,
+                    ram_data_size: collection_info_response.ram_data_size as usize,
+                    config: match collection_info_response.config {
+                        None => {
+                            return Err(Status::invalid_argument("Malformed CollectionConfig type"))
+                        }
+                        Some(config) => config.try_into()?,
+                    },
+                    payload_schema: collection_info_response
+                        .payload_schema
+                        .into_iter()
+                        .map(|(k, v)| (k, v.try_into().unwrap())) //TODO unwrap
```

Comment:
I did not find clean way to apply the `collect` method with the `HashMap` to transpose the `Result` type at the outer layer like it is done for a `Vec`.
Anyone got an idea?

Reply:
I have a similar case in my PR with HashMap and iter of results, it's in Raft State. I can link it here tomorrow.

Reply:
https://github.com/qdrant/qdrant/pull/509/files#diff-bfa448d463a5f39874fbdd86b9a0c9afc202a1e6d4dbcd1168af6eef2f7fe0a0R173-R175

Reply:
Done in https://github.com/qdrant/qdrant/pull/505/commits/a021682b0d2deee6074ca1b71dc2d99a36fba9af